mewarpx/mewarpx/sim_control.py

```diff
@@ -26,9 +28,16 @@ def __init__(self, total_steps, criteria=None):
             total_steps (int): Total number of steps to perform in the
                 simulation, when step == total_steps check_criteria returns
                 False
+            diag_steps (int): Steps between diagnostic output
             criteria (list): list of user defined functions or list of user
                 defined tuples(function, kwargs) that each return a True or
                 False value
+            checkpoint (bool): Whether or not simulation checkpoints should be
+                made
+            dump_period (int): Checkpoints will be created every
+                diag_steps*dump_period steps. If the dump_period is not given,
+                checkpoints will only be created when the simulation ends (
+                including interrupts due to a TERMINATE signal)
```

Comment:
To be clear, how does this end up interacting with the 'checkpoint now' signal action, that doesn't end the simulation? I know that's using the "dump at end of simulation" code path, but it's probably good to clarify that it can be invoked otherwise.

Reply:
Yeah the 'checkpoint now' signal won't work in `mewarpx` at present since it won't generate the corresponding fluxdiag checkpoint. Getting it to work will require another C++ variable to also track whether that signal was received with some more delicate logic to ensure that it is read before being reset for the next step.
I'm not very concerned about getting that to work for us though since I don't foresee us using it much at all. The SIGTERM handling though is blocking me from doing Perlmutter runs so getting that merged sooner rather than later would be nice.

Reply:
Ok, fine by me